wordpress-mobile/WordPress-iOS
# Reader: add P2 section

Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/15343
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/312

This adds a "P2s" section to the Reader. 
- It appears if you are following or are a user of a P2.
- It does not appear if you are not.

Known issue: If you are simply following a P2, the reader card header is not the [P2 header](https://github.com/wordpress-mobile/WordPress-iOS/pull/15315). This is due to the `is_wpforteams_site` flag missing from the site options for followed P2s. Resolution is pending Lighthouse issue https://github.com/Automattic/p2/issues/1542.

To test:

---
No P2s:
- Log in with an account that is not subscribed to any P2s.
- Verify the Reader does not have a `P2s` menu item.

<kbd><img width="400" alt="no_p2" src="https://user-images.githubusercontent.com/1816888/101226690-916d2700-3652-11eb-912d-e52bee33cafc.png"></kbd>

---
P2s:
- Log in with an account that is following/a user on a P2.
- Verify the Reader does have a `P2s` menu item.
- Due to the Known Issue mentioned above, the card headers will differ by role.
  - On a P2 that you are following, or are a Viewer, the header will be default. (The first card in the screenshot below.)
  - On a P2 where your role is higher that Viewer, the header will be P2. (The second card in the screenshot below.)

<kbd><img width="400" alt="p2s" src="https://user-images.githubusercontent.com/1816888/101227103-fa08d380-3653-11eb-88ba-d829b81ed81b.png"></kbd>

---
Roles reference, from https://wordpress.com/people/new/:

<kbd><img width="400" alt="roles" src="https://user-images.githubusercontent.com/1816888/101226980-8070e580-3653-11eb-978a-8c4357d81dd7.png"></kbd>


---
PR submission checklist:
